torch/_subclasses/fake_tensor.py

```diff
@@ -502,7 +504,7 @@ def new(self, *args, **kwargs):
         # when attempting to compute an output in meta, so
         # we compute the real tensor then convert to meta
         out_device = self.fake_device
-        with no_dispatch():
+        with no_dispatch(), in_kernel_invocation_manager(self.fake_mode):
```

Comment:
@eellison should we be applying `in_kernel_invocation_manager` for ALL no_dispatch brackets? It seems safer (as tensors will report that they're meta then.)

Reply:
I believe that's what I did originally, prior to updating the dispatch keys to show cuda. `new` is non-standard, and I didn't really delve deep into it, just got it work.